scripts/dd/dd-compare.py

```diff
@@ -0,0 +1,160 @@
+# Purpose: Takes in two PRISSMM data dictionaries and print out a simple comparison report
+# Author: Sage Bionetworks
+# Date: 29AUG2024
+
+import pandas as pd
+import synapseclient
+import argparse
+from io import StringIO
+
+# Log in 
+syn = synapseclient.Synapse()
+syn.login()
+
+# Set the default PRISSMM data dictionaries
+
+default1 = syn.get('syn52903784').path #v4.0.1
+default2 = syn.get('syn61600728').path #v4.1.4
+
+# Initialize arg parser
+parser = argparse.ArgumentParser(description="Compare two data dictionaries.")
+parser.add_argument('--df1', type=str, default=default1, help='File path to the older data dictionary CSV')
+parser.add_argument('--df2', type=str, default=default2, help='File path to the newer data dictionary CSV')
+
+# Parse the arguments
+args = parser.parse_args()
+
+entity1 = syn.get(args.df1)
+entity2 = syn.get(args.df2)
+
+# Function to clean up the multiline content while preserving specific rows
+def clean_multiline_content(file_content):
+    cleaned_lines = []
+    temp_line = ''
+    inside_quotes = False
+
+    for line in file_content:
+        if 'Change Type = Removed' in line:
+            cleaned_lines.append(line.strip())
+            continue
+        
+        if inside_quotes:
+            temp_line += ' ' + line.strip()
+            if line.count('"') % 2 != 0:  # Odd number of quotes indicates end of quoted section
+                inside_quotes = False
+                cleaned_lines.append(temp_line)
+                temp_line = ''
+        else:
+            if line.count('"') % 2 != 0:  # Odd number of quotes indicates start of quoted section
+                inside_quotes = True
+                temp_line = line.strip()
+            else:
+                cleaned_lines.append(line.strip())
+                
+    return cleaned_lines
+
+# Function to read and clean a CSV file, then load into a DataFrame
+def read_and_clean_csv(file_path):
+    with open(file_path, 'r') as file:
+        content = file.readlines()
+
+    # Clean the content
+    cleaned_content = clean_multiline_content(content)
+
+    # Create a temporary CSV file in memory
+    temp_csv = StringIO("\n".join(cleaned_content))
```

Comment:
Nice usage of StringIO here. Im surprised pandas doesn't have the functionality to read these things nicely.